libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -114,7 +113,25 @@ const getSharepointData = async (url) => {
   };
 };
 
-const getProjectExcelPath = (url) => url.pathname.replace('.json', '.xlsx');
+const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => {
+  try {
+    const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
+    if (!status.ok) throw new Error('Failed to fetch file path');
+    return (new URL((await status.json())?.preview?.url)).pathname;
+  } catch (e) {
+    throw new Error(`Error getting file path: ${e.message}`);
+  }
+};
```

Comment:
You already throw an error for the status, so it can be moved out of the try/catch
```suggestion
const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => {
  const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
  if (!status.ok) throw new Error('Failed to fetch file path');
  try {
    return (new URL((await status.json())?.preview?.url)).pathname;
  } catch (e) {
    throw new Error(`Error getting file path: ${e.message}`);
  }
};
```

Reply:
I took it one step further and removed the try/catch altogether. [The Task will catch any error](https://github.com/adobecom/milo/pull/2555/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR290) and show the message in the ui.